ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilder.go

```diff
@@ -160,12 +169,60 @@ func (pbuilder *PolicyBuilder) setValues(sourceMap map[string]interface{}, value
 	return sourceMap
 }
 
+func (pbuilder *PolicyBuilder) splitYamls(yamls []byte) ([][]byte, error) {
+	decoder := yaml.NewDecoder(bytes.NewReader(yamls))
+	var resources [][]byte
+
+	for {
+		var resIntf interface{}
+		err := decoder.Decode(&resIntf)
+
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		resBytes, err := yaml.Marshal(resIntf)
+
+		if err != nil {
+			return nil, err
+		}
+		resources = append(resources, resBytes)
+	}
+	return resources, nil
+}
+
+func (pbuilder *PolicyBuilder) getPolicy(name string, namespace string, resources []map[string]interface{}) utils.AcmPolicy {
+	acmPolicy := CreateAcmPolicy(name, namespace)
+	err := CheckNameLength(namespace, name)
+
+	if err != nil {
+		panic(err)
+	}
+	objTempArr := make([]utils.ObjectTemplates, 0)
+
+	for _, resourse := range resources {
+		objTempArr = append(objTempArr, CreateObjTemplates(resourse))
+	}
+	acmConfigPolicy := CreateAcmConfigPolicy(name)
+	acmConfigPolicy.Spec.ObjectTemplates = objTempArr
```

Comment:
The CreateAcmConfigPolicy method creates the acmConfigPolicy object. It would be a bit cleaner to pass the objTempArr into the method and let the method be responsible for fully constructing the object. That way the caller (this code here) doesn't have to know anything about the inner structure of acmConfigPolicy.

Reply:
done